src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/visualizers/RequestViewHTTP.java

```diff
@@ -285,9 +285,17 @@ public static Map<String, String[]> getQueryMap(String query) {
 
         Map<String, String[]> map = new HashMap<>();
         String[] params = query.split(PARAM_CONCATENATE);
+        int i = 0;
         for (String param : params) {
+            i++;
             String[] paramSplit = param.split("=");
-            String name = decodeQuery(paramSplit[0]);
+            String name = "";
```

Comment:
Is this OK? We use `name` further down to combine key/value-pairs with the same key.
In the old implementation we would not get to the point, where name is not initialized, now, we would get there with a default value of `""`.
Maybe use a `null` value of `name` to mean *no valid key*  found and use that to deduce the bool `noNameAndNoValue` later (see my other comment about double negations).